app/models/worldwide_organisation.rb

```diff
@@ -39,6 +39,10 @@ class WorldwideOrganisation < ApplicationRecord
 
   include PublishesToPublishingApi
 
+  attr_accessor :state
+
+  include Edition::AuditTrail
```

Comment:
To build upon @floehopper's [comment](https://github.com/alphagov/whitehall/pull/7574#issuecomment-1516578688) and @davidgisbey's [comment in Slack](https://gds.slack.com/archives/C02L13S214K/p1682011364360699?thread_ts=1682005365.826279&cid=C02L13S214K):

Hardcoding `state: "non-editionable"` seems like a bit of a hack – necessary to work around the fact that `Edition::AuditTrail` expects the object to have a `state`, but if we're honest it's probably a bit of a smell.

Instead, I'd suggest we avoid storing a `state` for non-editioned models altogether.

In fact, it looks like the `state` field is already optional. The [database schema allows NULL values](https://github.com/alphagov/whitehall/blob/a691d049a8f54a649b35a9a1bc4fa634c09d5895/db/schema.rb#L1082), and the [Version model](https://github.com/alphagov/whitehall/blob/ad86a5ea8d7538f787c71cb586adb1b771ea08d9/app/models/version.rb) doesn't specify validation rules on that field.

### Make `Edition::AuditTrail` more generic

We're starting to use `Edition::AuditTrail` for something slightly broader than what it was originally designed for. To date it's only been used on Edition-type models – so it has always been safe to assume they had a `state`.

Now that we're using it on non-Edition-type models, I'd suggest we consider making it more generic.

1. Rename `Edition::AuditTrail` to `AuditTrail`. I think we should be able to store this at `app/models/concerns/audit_trail.rb`.
2. Stop assuming that `state` is defined on the model – fall back to `nil` if it isn't present

```diff
  def record_create
    user = Edition::AuditTrail.whodunnit
+   state = self.try(:state)
    versions.create!(event: "create", user:, state:)
    alert!(user)
  end
```

### Beware other Edition-specific logic

I also want to point out the `Edition::AuditTrail#alert!` method which assumes it's dealing with an Edition-type model:

https://github.com/alphagov/whitehall/blob/73e1f38e0c64e505a6f1fbc5f54bb940447a5f05/app/models/edition/audit_trail.rb#L55-L59

We should consider this when making `AuditTrail` more generic.

Reply:
I wonder if those alerts could be shifted elsewhere, maybe into another concern `Edition::Alerts`

Reply:
Hmm 🤔 having taken a closer look, I think this might actually be dead code.

The `CO_NSS_WATCHKEEPER_EMAIL_ADDRESS` environment variable is not set in any of our live environments.

I'll double check with the team, but suspect the `alert!` functionality could be left alone in this PR, and a separate clean-up card created to remove it.

Reply:
That seems a fair point @jkempster34. Feel free to leave that alert functionality as is for now - we may well open a card to refactor it into a more appropriate place later.

Reply:
Even though ‘edition_published_by_monitored_user’ has the word edition in the method name, the only parameter that is passed is the ‘user’ and I think it’s possible/likely that whoever needs to be alerted about changes made by the special user might well want to be alerted about *any* changes made by that special user. I think the method is badly named.